### ISSUE TITLE
bitcoind takes 670% CPU usage at first sync

### ISSUE THREAD

**Issue by Mellnik** [bitcoin/bitcoin#6358]
and a unix load of *6.41* even though I only have a Quad Core CPU. It renders my machine fully unusable. I set it's nice value to 10 for now.

http://puu.sh/iIDtg/021875dd2a.png

**Comment by laanwj** [comment]
Which version?
What arguments / bitcoin.conf settings? You're sure you don't have `-gen=1` enabled?

You an try e.g. `-par=1` to reduce the number of script verification threads. By default it uses as many threads as there are cores available *minus one*, but it may be mis-estimating this.

Can you find a message like this in your debug.log:
```
2015-06-29 09:35:16 Using 5 threads for script verification
```

**Comment by Mellnik** [comment]
0.10.2

Config:
```
disablewallet=1
maxconnections=1000
timeout=10000
allowreceivebyip=1

bind=213.163.74.163:8333
bind=[2a00:1630:2:1901::b7c]:8333
rpcuser=bitcoinrpc
rpcpassword=xyz

addnode=node.bitcoin.xxx
addnode=x.jine.se
addnode=bitcoin.coinprism.com
addnode=btcnode1.evolyn.net
addnode=faucet.bitcoin.st

addnode=messier.bzfx.net
```
screen ./bitcoind -server

By now I rsynced the blockchain from my older node.

```
2015-06-30 21:05:40 Using 8 threads for script verification
```
Probably 8 because of HT.

**Comment by laanwj** [comment]
So it looks like `boost::thread::hardware_concurrency()`, which is used to determine the default concurrency, also counts 'fake' HT cores. It is a bit disappointing but I'm not sure what underlying APIs for querying this that operating systems expose in any case.

There is also [boost::thread::physical_concurrency()](http://www.boost.org/doc/libs/1_58_0/doc/html/thread/thread_management.html#thread.thread_management.thread.physical_concurrency) maybe it works better.

**Comment by sipa** [comment]
I'm marking this as fixed by #6361. Comment if this is not correct, please.
